### PR TITLE
HVG-619: Adyen Connect Payout, API integration

### PR DESCRIPTION
**Pull request by hedvigoscar**
### Checklist

- [ ] Functionality is covered by an integration test
- [X] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            DropIn.startPayment(this, response, dropInConfiguration)
+        }
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        when (intent?.getStringExtra(DropIn.RESULT_KEY)) {
+            ADYEN_RESULT_CODE_RECEIVED -> makeToast("TODO: Show success")
+            ADYEN_RESULT_CODE_CANCELLED -> makeToast("TODO: Show cancelled")
+            else -> {}
```
Comment (MathiasOhrstrom) [inline app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt:69]: Implement this, or is this for a later PR?

File: app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        finish()
+    }
```
Comment (MathiasOhrstrom) [inline app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt:76]: Cant we just add `android:noHistory="true"`? But I might have misunderstood when `onActivityResult()` is called

File: app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+abstract class AdyenConnectPayoutViewModel : ViewModel() {
+    protected val _payoutMethods = MutableLiveData<PaymentMethodsApiResponse>()
+    val payoutMethods: LiveData<PaymentMethodsApiResponse> = _payoutMethods
+}
+
+class AdyenConnectPayoutViewModelImpl(
+    private val repository: AdyenRepository
+) : AdyenConnectPayoutViewModel() {
```
Comment (MathiasOhrstrom) [inline app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt:97]: Move these to their own file

**Review by MathiasOhrstrom — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt (hunk 1 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt:69]: Later PR!

File: app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt (hunk 2 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/adyen/payout/AdyenConnectPayoutActivity.kt:76]: `onActivityResult` is called when the Adyen activity exits under certain circumstances. In this case, it's when the sheet is dismissed.

Good eye, though! This function contains a bug - it should only exit when `resultCode == Activity.RESULT_CANCELED`.

**Review by MathiasOhrstrom — APPROVED** [review]

**Pull request merged**